src/k8s/pkg/utils/certificate.go

```diff
@@ -26,3 +31,58 @@ func SplitIPAndDNSSANs(extraSANs []string) ([]net.IP, []string) {
 
 	return ipSANs, dnsSANs
 }
+
+func TLSClientConfigWithTrustedCertificate(remoteCert *x509.Certificate) (*tls.Config, error) {
```

Comment:
nit: Please add comments/explanation for public functions and some basic unittests. We want tiobe to be happy :)